### PR TITLE
Metrics 21 merge into master

### PR DESCRIPTION
**Pull request by ravi-varadarajan**

### CODE REVIEW

File: flow/util/genMetrics_21.py [hunk 1]
```diff
@@ -0,0 +1,488 @@
+#!/usr/bin/env python3
```
Comment (vvbandeira) [inline flow/util/genMetrics_21.py]: remove this file

**Review by vvbandeira — CHANGES_REQUESTED** [review]

File: flow/util/genMetrics_21.py (hunk 1 above)
Comment (ravi-varadarajan) [reply flow/util/genMetrics_21.py]: Can we remove this file as part of the next PR which will add the multi-clocks (ETD today).. This file is not being used by any of the scripts. It was used to run standalone for bigDoE that Seungwon was running

File: flow/util/genMetrics.py [hunk 2]
```diff
@@ -467,19 +418,19 @@ def extract_metrics(cwd, platform, design, flow_variant, output):
...
                        "Total +\S+ +\S+ +\S+ +(\S+) +\S+",
                        logPath+"/6_report.log")
 
-    extractTagFromFile("finish__area", metrics_dict,
+    extractTagFromFile("finish__design__instance__area", metrics_dict,
                       "^Design area (\S+) u\^2",
                        logPath+"/6_report.log")
 
-    extractTagFromFile("finish__util", metrics_dict,
+    extractTagFromFile("finish__design__instance__utilization", metrics_dict,
```
Comment (rovinski) [inline flow/util/genMetrics.py:433]: Why are there a bunch of keys with `design` in the name? Does that really add info? e.g. `finish__design__instance__utilization` vs. `finish__instance__utilization`

Comment (ravi-varadarajan) [reply flow/util/genMetrics.py:433]: In Metrics 2.1 format, we have a notion of "stage" and metrics category.  In this case "finish" is the stage  and "design" is the metrics category that hold all  physical information.  The current categories  are "design", "clock", "timing", "route" and "power".  The default stages are   run | init | synth | floorplan | globalplace | placeopt | detailedplace | cts | globalroute | detailedroute | final.   Stages can also be user defined like "user specific" snapshot

https://docs.google.com/spreadsheets/d/1Dbw8wj8005A__0hdKbIxiGfSAQUm36D9tdZi0AQZ7sY/edit#gid=1794314402

Comment (rovinski) [reply flow/util/genMetrics.py:433]: I see, thanks.

File: flow/designs/asap7/gcd/rules.json [hunk 3]
```diff
@@ -11,32 +11,32 @@
...
-            "field": "placeopt__area__instance__count",
+            "field": "placeopt__design__instance__stdcell__count",
             "value": 503,
             "compare": "<="
         },
         {
-            "field": "detailedroute__wirelength",
-            "value": 1455,
-            "compare": "<="
+            "field": "globalroute__timing__clock__slack",
```
Comment (maliberty) [inline flow/designs/asap7/gcd/rules.json:24]: I don't see a metric with this name in the spreadsheet.

File: flow/designs/intel22/aes/rules.json [hunk 4]
```diff
@@ -1,44 +1,9 @@
 {
     "rules": [
-        {
-            "field": "synth__area__stdcell__count",
-            "value": 28411,
-            "compare": "<="
-        },
         {
             "field": "constraints__clocks__count",
```
Comment (maliberty) [inline flow/designs/intel22/aes/rules.json:4]: Why is this reduced to one rule?

File: flow/designs/nangate45/swerv_wrapper/rules.json [hunk 5]
```diff
@@ -1,49 +1,14 @@
...
-        },
-        {
-            "field": "detailedroute__drc__error__count",
-            "value": 0,
-            "compare": "=="
-        },
-        {
-            "field": "finish__area",
-            "value": 2262308,
-            "compare": "<="
```
Comment (maliberty) [inline flow/designs/nangate45/swerv_wrapper/rules.json]: Why are these removed?

File: flow/designs/tsmc65lp/swerv_wrapper/metadata-base-ok.json [hunk 6]
```diff
@@ -1,99 +1,87 @@
 {
-  "run__flow__generate__date": "2021-05-13 08:34",
```
Comment (maliberty) [inline flow/designs/tsmc65lp/swerv_wrapper/metadata-base-ok.json:2]: Why is there no update to the rules file?

File: flow/util/genMetrics.py [hunk 7]
```diff
@@ -279,91 +275,63 @@ def extract_metrics(cwd, platform, design, flow_variant, output):
...
-                       "Inserted (\d+) input buffers",
-                       logPath+"/3_3_resizer.log")
-
-    extractTagFromFile("placeopt__area__outbuffer__count", metrics_dict,
-                       "Inserted (\d+) output buffers",
-                       logPath+"/3_3_resizer.log")
-
-    extractTagFromFile("placeopt__area__resize__count", metrics_dict,
-                       "Resized (\d+) instances",
-                       logPath+"/3_3_resizer.log")
```
Comment (maliberty) [inline flow/util/genMetrics.py:310]: are we not tracking these anymore?

File: flow/designs/nangate45/swerv_wrapper/rules.json (hunk 5 above)
Comment (vvbandeira) [reply flow/designs/nangate45/swerv_wrapper/rules.json]: This is related to the [OR-998](https://openroad.atlassian.net/browse/OR-998) -- i.e., the design was failing and we could not generate a new baseline. Since Cherry already fixed this, @ravi-varadarajan could you run this design again (maybe using OpenROAD master) to get an ok and rules file? Or you can commit using [these artifacts](https://jenkins.openroad.tools/blue/organizations/jenkins/OpenROAD-flow-scripts-Nightly-Public/detail/OpenROAD-flow-scripts-Nightly-Public/45/artifacts) from CI.

File: flow/designs/intel22/aes/rules.json (hunk 4 above)
Comment (vvbandeira) [reply flow/designs/intel22/aes/rules.json:4]: We probably need a new baseline ok file to generate the rules file with the new naming.

File: flow/util/genMetrics_21.py (hunk 1 above)
Comment (ravi-varadarajan) [reply flow/util/genMetrics_21.py]: Removed this file

File: flow/designs/asap7/gcd/rules.json (hunk 3 above)
Comment (maliberty) [reply flow/designs/asap7/gcd/rules.json:24]: Agreed to rename in subsequent PR

File: flow/util/genMetrics.py (hunk 7 above)
Comment (maliberty) [reply flow/util/genMetrics.py:310]: not currently tracked or in metrics 2.1.  Could be reinstated later.

**Review by vvbandeira — DISMISSED** [review]

File: flow/designs/tsmc65lp/swerv_wrapper/metadata-base-ok.json (hunk 6 above)
Comment (vvbandeira) [reply flow/designs/tsmc65lp/swerv_wrapper/metadata-base-ok.json:2]: fixed on 896c2d1

File: flow/designs/nangate45/swerv_wrapper/rules.json (hunk 5 above)
Comment (vvbandeira) [reply flow/designs/nangate45/swerv_wrapper/rules.json]: fixed on 2e36130

**Comment by ravi-varadarajan** [comment]
Hi Matt  Can you approve this PR.  Once this is merged there are a couple of incremental changes that I want to submit namely for Stephano and the update for multiple clocks.  thanks

**Review by maliberty — APPROVED** [review]
